src/modules/MovieCard/MovieCard.tsx

```diff
@@ -36,12 +37,13 @@ export const MovieCard = ({
   },
   remove,
   edit,
+  changeMode,
 }: IMovieCard) => {
   const [isOpen, setIsOpen] = useState(false);
 
 
   return (
-    <MovieCardWrapper onClick={() => isOpen && setIsOpen(false)}>
+    <MovieCardWrapper onClick={() => {changeMode(); isOpen && setIsOpen(false)}}>
```

Comment:
difficult to read
```suggestion
    <MovieCardWrapper onClick={() => {
       changeMode(); 
       if (isOpen) {
         setIsOpen(false);
       }
     }}>
```

also you can try to make it throw `useEffect`
```
useEffect(() => {
  if(isOpen) {
     setIsOpen(true);
  }
});

...

<MovieCardWrapper onClick={changeMode} />
```